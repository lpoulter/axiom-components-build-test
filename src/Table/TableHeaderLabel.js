import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classnames from 'classnames';
import Base from '../Base/Base';
import TextIcon from '../Typography/TextIcon';

export default class TableHeaderLabel extends Component {
  static propTypes = {
    /** Table header label */
    children: PropTypes.node,
    className: PropTypes.node,
    /** Expands table column (Incompatible with 'width') */
    grow: PropTypes.bool,
    /** Makes label clickable */
    onClick: PropTypes.func,
    /** Shrinks table column (Incompatible with 'width') */
    shrink: PropTypes.bool,
    /** Selects the column and sets the sort direction */
    sortDirection: PropTypes.oneOf(['ascending', 'descending']),
    /** Indicates the column is sortable if false it will remove the cloaked icon
    indicating sort direction */
    sortable: PropTypes.bool,
    /** Set text-align */
    textAlign: PropTypes.oneOf(['left', 'right', 'centre']),
    /** Set percentage column width (Incompatible with 'grow' and 'shink') */
    width: PropTypes.number,
  };

  static defaultProps = {
    textAlign: 'left',
    sortable: true,
  }

  render() {
    const {
      children,
      className,
      grow,
      onClick,
      shrink,
      sortDirection,
      textAlign,
      sortable,
      width,
      ...rest
    } = this.props;

    const classes = classnames(
      'ax-table__header-label',
      `ax-table__header-label--align-${textAlign}`,
      {
        'ax-table__header-label--grow': grow && !width,
        'ax-table__header-label--selected': sortDirection !== undefined,
        'ax-table__header-label--shrink': shrink && !width,
      },
      className
    );

    const styles = width && {
      width: `${Math.max(0, Math.min(width, 100))}%`,
    };

    return (
      <Base { ...rest } Component="th" className={ classes } style={ styles }>
        <button
            className="ax-table__header-button"
            disabled={ !onClick }
            onClick={ onClick }>
          { children }

          { sortable && ( <TextIcon
              cloak={ sortDirection === undefined }
              name={ sortDirection === 'descending' ? 'triangle-down' : 'triangle-up' }
              spaceLeft={ textAlign === 'left' ? 'x2' : undefined }
              spaceRight={ textAlign === 'right' ? 'x2' : undefined } />
          ) }
        </button>
      </Base>
    );
  }
}
