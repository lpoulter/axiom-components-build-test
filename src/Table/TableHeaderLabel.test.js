import React from 'react';
import renderer from 'react-test-renderer';
import TableHeaderLabel from './TableHeaderLabel';

const getComponent = props => renderer.create(
  <TableHeaderLabel { ...props }>Test</TableHeaderLabel>
);

describe('TableHeaderLabel', () => {
  it('renders', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with grow', () => {
    const component = getComponent({ grow: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with sortDirection', () => {
    const component = getComponent({ sortDirection: 'ascending' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with shrink', () => {
    const component = getComponent({ shrink: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with textAlign', () => {
    const component = getComponent({ textAlign: 'right' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with sortable', () => {
    const component = getComponent({ sortable: false });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with width', () => {
    const component = getComponent({ width: 20 });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('width overrides grow', () => {
    const component = getComponent({ width: 20, grow: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('width overrides shrink', () => {
    const component = getComponent({ width: 20, shrink: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
