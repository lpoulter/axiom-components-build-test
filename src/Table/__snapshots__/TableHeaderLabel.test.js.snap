// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeaderLabel renders 1`] = `
<th
  className="ax-table__header-label ax-table__header-label--align-left"
>
  <button
    className="ax-table__header-button"
    disabled={true}
  >
    Test
    <svg
      className="ax-icon ax-icon--triangle-up ax-icon--inline ax-icon--space-left-x2 ax-cloak"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
      style={
        Object {
          "height": "1rem",
          "width": "1rem",
        }
      }
      viewBox="0 0 16 16"
    />
  </button>
</th>
`;

exports[`TableHeaderLabel renders with grow 1`] = `
<th
  className="ax-table__header-label ax-table__header-label--align-left ax-table__header-label--grow"
>
  <button
    className="ax-table__header-button"
    disabled={true}
  >
    Test
    <svg
      className="ax-icon ax-icon--triangle-up ax-icon--inline ax-icon--space-left-x2 ax-cloak"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
      style={
        Object {
          "height": "1rem",
          "width": "1rem",
        }
      }
      viewBox="0 0 16 16"
    />
  </button>
</th>
`;

exports[`TableHeaderLabel renders with shrink 1`] = `
<th
  className="ax-table__header-label ax-table__header-label--align-left ax-table__header-label--shrink"
>
  <button
    className="ax-table__header-button"
    disabled={true}
  >
    Test
    <svg
      className="ax-icon ax-icon--triangle-up ax-icon--inline ax-icon--space-left-x2 ax-cloak"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
      style={
        Object {
          "height": "1rem",
          "width": "1rem",
        }
      }
      viewBox="0 0 16 16"
    />
  </button>
</th>
`;

exports[`TableHeaderLabel renders with sortDirection 1`] = `
<th
  className="ax-table__header-label ax-table__header-label--align-left ax-table__header-label--selected"
>
  <button
    className="ax-table__header-button"
    disabled={true}
  >
    Test
    <svg
      className="ax-icon ax-icon--triangle-up ax-icon--inline ax-icon--space-left-x2 ax-cloak ax-cloak--visible"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
      style={
        Object {
          "height": "1rem",
          "width": "1rem",
        }
      }
      viewBox="0 0 16 16"
    />
  </button>
</th>
`;

exports[`TableHeaderLabel renders with sortable 1`] = `
<th
  className="ax-table__header-label ax-table__header-label--align-left"
>
  <button
    className="ax-table__header-button"
    disabled={true}
  >
    Test
  </button>
</th>
`;

exports[`TableHeaderLabel renders with textAlign 1`] = `
<th
  className="ax-table__header-label ax-table__header-label--align-right"
>
  <button
    className="ax-table__header-button"
    disabled={true}
  >
    Test
    <svg
      className="ax-icon ax-icon--triangle-up ax-icon--inline ax-icon--space-right-x2 ax-cloak"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
      style={
        Object {
          "height": "1rem",
          "width": "1rem",
        }
      }
      viewBox="0 0 16 16"
    />
  </button>
</th>
`;

exports[`TableHeaderLabel renders with width 1`] = `
<th
  className="ax-table__header-label ax-table__header-label--align-left"
  style={
    Object {
      "width": "20%",
    }
  }
>
  <button
    className="ax-table__header-button"
    disabled={true}
  >
    Test
    <svg
      className="ax-icon ax-icon--triangle-up ax-icon--inline ax-icon--space-left-x2 ax-cloak"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
      style={
        Object {
          "height": "1rem",
          "width": "1rem",
        }
      }
      viewBox="0 0 16 16"
    />
  </button>
</th>
`;

exports[`TableHeaderLabel width overrides grow 1`] = `
<th
  className="ax-table__header-label ax-table__header-label--align-left"
  style={
    Object {
      "width": "20%",
    }
  }
>
  <button
    className="ax-table__header-button"
    disabled={true}
  >
    Test
    <svg
      className="ax-icon ax-icon--triangle-up ax-icon--inline ax-icon--space-left-x2 ax-cloak"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
      style={
        Object {
          "height": "1rem",
          "width": "1rem",
        }
      }
      viewBox="0 0 16 16"
    />
  </button>
</th>
`;

exports[`TableHeaderLabel width overrides shrink 1`] = `
<th
  className="ax-table__header-label ax-table__header-label--align-left"
  style={
    Object {
      "width": "20%",
    }
  }
>
  <button
    className="ax-table__header-button"
    disabled={true}
  >
    Test
    <svg
      className="ax-icon ax-icon--triangle-up ax-icon--inline ax-icon--space-left-x2 ax-cloak"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
      style={
        Object {
          "height": "1rem",
          "width": "1rem",
        }
      }
      viewBox="0 0 16 16"
    />
  </button>
</th>
`;
